pkg/verifier/cosign/cosign.go

```diff
@@ -103,12 +115,20 @@ func (f *cosignVerifierFactory) Create(_ string, verifierConfig config.VerifierC
 		return nil, re.ErrorCodeConfigInvalid.WithComponentType(re.Verifier).WithPluginName(verifierName)
 	}
 
+	trustPolicies, err := CreateTrustPolicies(config.TrustPolicies, verifierName)
+	if err != nil {
+		return nil, err
+	}
+
 	return &cosignVerifier{
 		name:             verifierName,
 		verifierType:     config.Type,
 		artifactTypes:    strings.Split(config.ArtifactTypes, ","),
 		nestedReferences: config.NestedReferences,
 		config:           config,
+		isLegacy:         config.KeyRef != "" || config.RekorURL != "",
```

Comment:
have a check for if both legacy and new trust policies are configured. throw error